Formula/kde-kplotting.rb

```diff
@@ -0,0 +1,64 @@
+class KdeKplotting < Formula
+  desc "Lightweight plotting framework"
+  homepage "https://api.kde.org/frameworks/kplotting/html/index.html"
+  url "https://download.kde.org/stable/frameworks/5.79/kplotting-5.79.0.tar.xz"
+  sha256 "c67238fb1395197b2ee2a65f327458132106df6110765952bd979b11cfbb5c51"
+  license all_of: [
+    "GPL-2.0-or-later",
+    "LGPL-2.0-or-later",
+  ]
+  head "https://invent.kde.org/frameworks/kplotting.git"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "doxygen" => :build
+  depends_on "graphviz" => :build
+  depends_on "kde-extra-cmake-modules" => [:build, :test]
+  depends_on "ninja" => :build
+
+  depends_on "qt@5"
+
+  def install
+    args = std_cmake_args
+    args << "-D BUILD_QCH=ON"
+    args << "-D BUILD_TESTING=OFF"
+    args << "-D BUILD_TESTS=OFF"
+    args << "-D BUILD_UNITTESTS=OFF"
+    args << "-D CMAKE_INSTALL_BUNDLEDIR=#{bin}"
+    args << "-D KDE_INSTALL_LIBDIR=lib"
+    args << "-D KDE_INSTALL_PLUGINDIR=lib/qt5/plugins"
+    args << "-D KDE_INSTALL_QMLDIR=lib/qt5/qml"
+    args << "-D KDE_INSTALL_QTPLUGINDIR=lib/qt5/plugins"
+
+    mkdir "build" do
+      system "cmake", "..", "-G", "Ninja", *args
+      system "ninja"
+      system "ninja", "install"
+      prefix.install "install_manifest.txt"
+    end
+
+    pkgshare.install "autotests"
+    pkgshare.install "examples"
+  end
+
+  test do
+    (testpath/"CMakeLists.txt").write <<~EOS
+      include(FeatureSummary)
+      find_package(ECM NO_MODULE)
+      set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} "#{pkgshare}/cmake")
+      find_package(Qt5Widgets ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+
+      add_subdirectory(autotests)
+      add_subdirectory(examples)
+    EOS
+
+    cp_r (pkgshare/"autotests"), testpath
+    cp_r (pkgshare/"examples"), testpath
+
+    args = std_cmake_args
+    args << "-DQt5_DIR=#{Formula["qt@5"].opt_lib/"cmake/Qt5"}"
+    args << "-DQt5Widgets_DIR=#{Formula["qt@5"].opt_lib/"cmake/Qt5Widgets"}"
+
+    system "cmake", testpath.to_s, *args
+    system "make"
```

Comment:
You're compiling but not actually testing anything that's built. That's not really in keeping with `test do` blocks. I'm not sure why this is more of a test than "did `install` succeed"?

Reply:
> I'm not sure why this is more of a test than "did install succeed"?

Well,`kplotting` is a library, so it tests whether it actually is possible to successfully build some code linking to it and using its API. I'd say that's much more thorough than "did the library itself install successfully"? 

The example included with the code is a QApplication, i.e. a GUI app, and none of the [built-in ](https://doc.qt.io/qt-5/qapplication.html#QApplication) switches are CLI.

Lastly, I followed the existing examples that do just that:
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/kde-ki18n.rb
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/kde-karchive.rb
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/kde-kdoctools.rb

Reply:
> Well,`kplotting` is a GUI library, so that test checks whether it actually is possible to successfully build some code linking to it _and_ using its API. I'd say that's much more thorough than "did the library itself install successfully"?

The test is opaque, though, compared to an inline C++ file in the `test do` block.

> Lastly, I followed the existing examples that do just that:
> 
> * https://github.com/Homebrew/homebrew-core/blob/master/Formula/kde-ki18n.rb
> * https://github.com/Homebrew/homebrew-core/blob/master/Formula/kde-karchive.rb
> * https://github.com/Homebrew/homebrew-core/blob/master/Formula/kde-kdoctools.rb

I would have left the same comments if I reviewed these.